orchestration/task_dispatcher.py

```
@@ -1085,6 +1324,43 @@ def create_dynamic_agent(self, agent_spec: dict) -> bool:
+            mock_agent_dir_name = os.environ.get("MOCK_AGENT_DIR", "mock_agents")
+            mock_dir = Path(tempfile.gettempdir()) / mock_agent_dir_name
+            mock_dir.mkdir(parents=True, exist_ok=True)
+            mock_path = mock_dir / "mock_claude.sh"
+
+            # Simple shim that echoes the call for logging and exits successfully
+            script_contents = """#!/usr/bin/env bash
+echo "[mock claude] $@"
+exit 0
+"""
+            Path.write_text(mock_path, script_contents, encoding="utf-8")
+            os.chmod(mock_path, 0o755)
```

Ruff (0.14.0)

`os.chmod` setting a permissive mask `0o755` on file or directory

(S103)